src/kernel/hardware/rtc.c

```diff
@@ -0,0 +1,91 @@
+#include <rtc.h>
+#include <io.h>
+#include <logging.h>
+
+uint64_t daysPerMonth[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
+
+uint8_t read_rtc_register(uint8_t port_number) {
+    outportb(CMOS_ADDRESS_REGISTER, port_number);    
+    return inportb(CMOS_DATA_REGISTER);
+}
+
+uint64_t read_rtc_time() {
+    // Yes is nearly identical to the one in northport! (i implemented it there first and then ported it to Dreamos! :) 
+
+    while ( is_rtc_updating() );
+    uint8_t rtc_register_statusB = read_rtc_register(StatusRegisterB);
+    bool is_24hours = rtc_register_statusB & 0x02;
+    bool is_binary = rtc_register_statusB & 0x04;
+
+    uint8_t seconds = read_rtc_register(Seconds);
+    uint8_t minutes = read_rtc_register(Minutes);
+    uint8_t hours = read_rtc_register(Hours);
+    uint8_t year = read_rtc_register(Year);
+    uint8_t dayofmonth = read_rtc_register(DayOfMonth);
+    uint8_t month = read_rtc_register(Month);
+    uint8_t last_seconds; 
+    uint8_t last_minutes;
+    uint8_t last_hours;
+    uint8_t last_year;
+    uint8_t last_dayofmonth;
+    uint8_t last_month;
+    do {
+        last_seconds = seconds;
+        last_minutes = minutes;
+        last_hours = hours;
+        last_year = year;
+        last_dayofmonth = dayofmonth;
+        last_month = month;
+
+        while ( is_rtc_updating() );
+        seconds = read_rtc_register(Seconds);
+        minutes = read_rtc_register(Minutes);
+        hours = read_rtc_register(Hours);
+        year = read_rtc_register(Year);
+        dayofmonth = read_rtc_register(DayOfMonth);
+        month = read_rtc_register(Month);
+    } while ( last_seconds == seconds && last_minutes == minutes && last_hours == hours && last_year == year &&  last_dayofmonth == dayofmonth && last_month == month);
+
+    if ( !is_binary) {
+        hours = convert_bcd_to_binary(hours);
+        seconds = convert_bcd_to_binary(seconds);
+        minutes = convert_bcd_to_binary(minutes);
+        year = convert_bcd_to_binary(year);
+        month = convert_bcd_to_binary(month);
+        dayofmonth = convert_bcd_to_binary(dayofmonth);
+    }
+
+    uint64_t yearsSinceEpoch = (BASE_CENTURY + year) - 1970; // Let's count the number of years passed since the Epoch Year: (1970)
```

Comment:
I'd say the same things I said about all these local variables in the northport PR:
There's a lot of variables that are initialized inline, if they are are marked as `const` it may allow the compiler to optimize better. 
I would also probably collapse line 60 into the if check below it, rather than having that as a separate variable.